sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -35,6 +36,7 @@ import org.apache.spark.util.collection.OpenHashSet
  *
  * TODO Clean up the metadata files periodically
  */
+@ThreadSafe
```

Comment:
i dont know about this. Could you explain briefly how does this annotation help?